prometheus_client/metrics.py

```diff
@@ -382,18 +383,25 @@ def _metric_init(self) -> None:
 
     def inc(self, amount: float = 1) -> None:
         """Increment gauge by the given amount."""
+        if self._multiprocess_mode in self._MOST_RECENT_MODES:
```

Comment:
I wonder if instead of having to do multiple string comparisons in each call for inc/dec we should set this as `_is_most_recent` or something like that on the gauge at init time?

Reply:
Done.